pkg/app/piped/cloudprovider/ecs/client.go

```diff
@@ -177,5 +174,10 @@ func (c *client) ServiceExists(ctx context.Context, clusterName string, services
 		}
 		return false, err
 	}
-	return true, nil
+	for _, service := range output.Services {
+		if *service.ServiceName == serviceName && *service.Status == "ACTIVE" {
+			return true, nil
+		}
+	}
+	return false, nil
```

Comment:
as for [executor/ecs/ecs.go#133](https://github.com/pipe-cd/pipe/pull/1725/files#diff-0ced6421324def8505cb6dadaa49f469d0855ef22437ba6d8ad7f5279a61b2e4R133), it looks like we're going to create a new service in case this check returns `found == false` but from this line, looks like this false will be returned on `service.Status` not equal `ACTIVE` only, which leads to duplicate create 👀  
please correct me if I'm wrong 🙏

Reply:
Note: In case of cluster's existing `serviceName` is set to `inactive` status, it's safe to recreate the service with the same `serviceName`.
cc @TaKO8Ki

Reply:
Done with it!